keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -363,13 +388,131 @@ Other components:
+1. Decrease memory Requests while increasing memory Limits.
+1. CPU: increase Requests, decrease Limits, Memory: increase Requests, decrease Limits.
+1. CPU: decrease Requests, increase Limits, Memory: decrease Requests, increase Limits.
+
+E2E tests for Burstable class single container Pod that specifies CPU only:
+1. Increase, decrease CPU - Requests only.
+1. Increase, decrease CPU - Limits only.
+1. Increase, decrease CPU - both Requests & Limits.
+
+E2E tests for Burstable class single container Pod that specifies memory only:
+1. Increase, decrease memory - Requests only.
+1. Increase, decrease memory - Limits only.
+1. Increase, decrease memory - both Requests & Limits.
+
+E2E tests for Guaranteed class Pod with three containers (c1, c2, c3):
+1. Increase CPU & memory for all three containers.
+1. Decrease CPU & memory for all three containers.
+1. Increase CPU, decrease memory for all three containers.
+1. Decrease CPU, increase memory for all three containers.
+1. Increase CPU for c1, decrease c2, c3 unchanged - no net CPU change.
+1. Increase memory for c1, decrease c2, c3 unchanged - no net memory change.
+1. Increase CPU for c1, decrease c2 & c3 - net CPU decrease for Pod.
+1. Increase memory for c1, decrease c2 & c3 - net memory decrease for Pod.
+1. Increase CPU for c1 & c3, decrease c2 - net CPU increase for Pod.
+1. Increase memory for c1 & c3, decrease c2 - net memory increase for Pod.
+
+### Resource Quota and Limit Ranges
+
+Setup a namespace with ResourceQuota and a single, valid Pod.
+1. Resize the Pod within resource quota - CPU only.
+1. Resize the Pod within resource quota - memory only.
+1. Resize the Pod within resource quota - both CPU and memory.
+1. Resize the Pod to exceed resource quota - CPU only.
+1. Resize the Pod to exceed resource quota - memory only.
+1. Resize the Pod to exceed resource quota - both CPU and memory.
+
+Setup a namespace with min and max LimitRange and create a single, valid Pod.
+1. Increase, decrease CPU within min/max bounds.
+1. Increase CPU to exceed max value.
+1. Decrease CPU to go below min value.
+1. Increase memory to exceed max value.
+1. Decrease memory to go below min value.
+
+### Resize Policy Tests
+
+Setup a guaranteed class Pod with two containers (c1 & c2).
+1. No resize policy specified, defaults to NoRestart. Verify that CPU and
+   memory are resized without restarting containers.
+1. NoRestart (cpu, memory) policy for c1, RestartContainer (cpu, memory) for c2.
+   Verify that c1 is resized without restart, c2 is restarted on resize.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 CPU only,
+   verify container is resized without restart.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 memory only,
+   verify container is resized with restart.
+1. NoRestart cpu, RestartContainer memory policy for c1. Resize c1 CPU & memory,
+   verify container is resized with restart.
+
+### Negative Tests
+TBD
+
```

Comment:
Given this touches a field involved in pod spec, pod template spec, and workload controllers, we need tests to make sure introduction of this does not cause workloads to redeploy on API server upgrade (e.g. https://github.com/kubernetes/kubernetes/issues/78633); tests that look something like what is described in https://github.com/kubernetes/kubernetes/issues/78904, and which are actually run